integrations/nodejs/javascript_async_eventloop.hpp

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+
+// This header file implements `JSAsyncEventLoopImpl`, a poor man's "full event loop" implementation that is based on
+// the very `JSAsync` / `AsyncWorker`. The implementation is suboptimal because it spawns a new `AsyncWorker` on every
+// new in-and-out call from within and outside the JavaScript event loop.
+//
+// The imprortant thing is that the semantics is right: the very thread-local singleton for JS environment is respected,
+// and the user code reads as it should.
+//
+// NOTE(dkorolev): I expect to get back to this code once performance considerations become important.
+
+#include "javascript_async.hpp"
+#include "javascript_function_cont.hpp"  // For `LambdaSignatureExtractor`.
+
+namespace current {
+namespace javascript {
+namespace impl {
+
+class JSAsyncEventLoopImpl final {
+ private:
+  mutable std::mutex mutex_;
+  mutable std::condition_variable cv_;
+
+  std::function<bool()> next_function_;
+  bool last_call_result_ = true;  bool terminated_ = false;
```

Comment:
Missing \n

Reply:
Oh, I must have introduced it during my final manual review cycle, as `make indent` fixed it. Good eye! :+1: